### DISCUSSION TITLE
Gemeinsames Profil für "Windows PowerShell" und "PowerShell

### DISCUSSION THREAD

**Discussion by costeffl** [TobiasPSP/OReilly#20] [Allgemeines]
Hallo zusammen,

Beschäftige mich seit ein paar Tagen mit Powershell mit Unterstützung des Buches Powershell 7 (O’Reilly) Habe jetzt Windows PwShell ise und Powershell 7.3.7 in der Mangel. Weiterhin muß ich hinzufügen das ich absoluter Neuling bin. Nun habe ich mit dem Script 1.16 aus dem Buch versucht eine Hardlink für 7.3. zu erstellen. Dies gelingt auch soweit. Leider lässt sich der Hardlink nicht erfolgreich aufrufen bzw. arbeitet nicht richtig. Nach meiner rudimentären Analys habe ich flgd. Passage in Verdacht. 

{
    # Pfad steht bei Windows PowerShell schon in $profile:
    $winprofile = $profile.CurrentUserAllHosts
    # $coreprofile =  $profile -replace '\WindowsPowerShell\','\\PowerShell\\'
    $coreprofile =  $profile -replace '\\WindowsPowerShell\\','\PowerShell\'
    
}

Es wird dann diese Profil Datei erzeugt. 

„C:\Users\Triumphator\Documents\PowerShell\Microsoft.PowerShellISE_profile.ps1“

Mit dieser Datei wird das Profil nicht ausgeführt. Wenn ich jedoch die Datei umbenenne funktioniert das.  

„C:\Users\Triumphator\Documents\PowerShell\profile.ps1“

Ich habe auch versucht herauszubekommen was genau dieser Befehl macht?

Ich hoffe ich bin einigermassen klar rübergekommen.

Grüße….

**Comment by TobiasPSP** [comment]
Hallo costeffl ;-)

Willkommen an Bord! 
Der Hardlink wurde korrekt angelegt, allerdings gilt die Profildatei nur für die ISE, nicht für die Konsole:

ISE-spezifisches Profil:

C:\Users\Triumphator\Documents\PowerShell\Microsoft.PowerShellISE_profile.ps1

generisches Profil:

C:\Users\Triumphator\Documents\PowerShell\profile.ps1

Ich gehe davon aus, dass Du die Wirksamkeit der Profildatei mit der PowerShell-Konsole oder VSCode überprüft hast.

$profile liefert immer das *persönliche* Profil für den *aktuellen* PowerShell-Host.
Über $profile.CurrentUserAllHosts bekommst Du das allgemeine (hostunspezifische) Profil.

Da es im Code indes heißt:

```powershell
$winprofile = $profile.CurrentUserAllHosts
```

müsste der erzeugte Dateiname eigentlich "profile.ps1" und nicht "Microsoft.PowerShellISE_profile.ps1" heißen.
Könntest Du die oben genannte Zeile noch mal überprüfen?

**Reply by costeffl** [reply]
Hallo Tobias,
danke für die Antwort. Dachte schon wegen des zeitlichen Alters des Forums keiner mehr hinschaut. Wie gesagt, habe gerade mit pwsh angefangen. Werde deinen Tip mal verinnerlichen. Melde mich wieder. Kann etwas dauern... ;-(

**Reply by costeffl** [reply]
so da bin ich nochmal. Alles gut... funktioniert alles. Danke. Mir war nicht klar das ISE eine eigene Console ist....

**Reply by TobiasPSP** [reply]
Bestens!
